diff --git a/build/node.js b/build/node.js
index 5fed586f2ce61db9921b663e8c3fdf445c21689e..fc9dce86e2e721a353bd5d29732c873465c4c20c 100644
--- a/build/node.js
+++ b/build/node.js
@@ -361,12 +361,15 @@ If you like to run a local browser session make sure to pick from one of the fol
       port
     });
   } else if (isFirefox(caps.browserName)) {
-    const { executablePath } = await setupPuppeteerBrowser(cacheDir, caps);
-    caps["moz:firefoxOptions"] = deepmerge(
-      { binary: executablePath },
-      caps["moz:firefoxOptions"] || {}
-    );
-    delete caps.browserVersion;
+    const isAndroid = Boolean(caps["moz:firefoxOptions"]?.androidPackage);
+    if (!isAndroid) {
+      const { executablePath } = await setupPuppeteerBrowser(cacheDir, caps);
+      caps["moz:firefoxOptions"] = deepmerge(
+        { binary: executablePath },
+        caps["moz:firefoxOptions"] || {}
+      );
+      delete caps.browserVersion;
+    }
     const { binary, ...geckodriverOptions } = caps["wdio:geckodriverOptions"] || {};
     if (binary) {
       geckodriverOptions.customGeckoDriverPath = binary;
